### ISSUE TITLE
Increase the Earmuff Mode Transition End Maximum

### ISSUE THREAD

**Issue by Ryn-Fox** [Yellow-Dog-Man/Resonite-Issues#3053]
### Is your feature request related to a problem? Please describe.

As somebody with severe ADHD who has difficulty in filtering out multiple conversations, I find it limiting that Earmuff Mode only allows for tuning such a small falloff transition range before it reaches the static Volume Attenuation value.

### Describe the solution you'd like

**I'd like the Transition End setting to have a much higher maximum value so I can more specifically tune a custom falloff for others' voices.

I'd additionally like if the Distance, Transition Min, and Transition Max more clearly specified the units they're using. I'm assuming all three represent Meters, but it's hard to know.**

Right now I find that if people are spread out, I can't quite create my ideal configuration where those right next to and in front of me are full volume, those outside of my immediate "bubble" (Earmuff Distance set to roughly 6 meters with 90% directionality and a 45 degree angle) are reduced in volume, and those much further away (let's say roughly 15+ meters) are completely muted. Additionally, I find that even with the Transition Min set to 0 and the Transition Max set to its maximum value of 2, the transition from full to minimum volume feels too abrupt.

Unless I'm doing something very wrong, I believe it's because the Transition End only allows for a 2 meter "window" for voices to fall off once they're beyond the configured "earmuff bubble".

**An alternative to increasing the slider max could be allowing custom typed values to override the slider, similar to how blendshapes work.**

### Describe alternatives you've considered

I'm very much open to learning if I'm maybe just misunderstanding some of the settings available to me. 

I could also see more general settings for voice falloff distance and linear/logarithmic falloff tuning being available.

### Additional Context

![Image](https://github.com/user-attachments/assets/a8f2370f-92fa-4242-8ce0-f5e320ff5da1)

### Requesters

Resonite: Ryn Fluff
Discord: RynFluff

**Comment by shiftyscales** [comment]
Seeking input from @Frooxius who had implemented this feature.

**Comment by Frooxius** [comment]
Are you essentially wanting the transition area to be a lot longer? Meaning that the user's voice will attenuate over much longer distance? Do you have a rough idea how much longer you want the transition to be?

We could also increase the overall distance too.

The units part is a bit odd one - it's sort of meters. It's in the game units, which are dimensionless. At unit scale they correspond to meters, but will depend on your scale.

**Comment by shiftyscales** [comment]
From experience- at the moment earmuffs don't seem to account for user scale @Frooxius - see #2230.

**Comment by Ryn-Fox** [comment]
@Frooxius You have the right idea, yes. I'd like the transition area to be a lot longer. I think 15 meters would be a pretty good "Transition Range". 

What would be the difference between changing that and the "overall distance"?

**Comment by Frooxius** [comment]
Hmm... 15 meters seems really huge for transition area. 

The transition area is essentially transition from where the earmuff is active to where it's not. Generally the idea is that it's relatively small transition area, because you want people to get ear-muffed quickly, otherwise it won't be as effective.

The main purpose for it is so the user's volume doesn't just instantly drop or rise as they cross the ear muff area.

The overall distance is the distance where the users are not earmuffed - they still have the normal distance attenuation, so they should sound quieter if they're further away. 

The way the system is designed is that you'd adjust the actual distance (without touching the transition area) to quickly include more or less people into the earmuffed region. We could increase that instead.

I'm a bit vary of increasing the transition range to such a big number, because it's not really meant to be for adjusting the actual distance where you can hear people, but more to give the earmuff area some "softness" around the edges.

**Comment by shiftyscales** [comment]
It sounds like it may be partially a misunderstanding in terminology then? "Transition Range" sounds like "the distance across which the audio will go from full volume to fully attenuated".

**Comment by Ryn-Fox** [comment]
I understand that it's not the original intent, but it fills a significant accessibility niche for me.


The common situation for me that this would solve is when I'm having a conversation with one or two people who are within 2 meters of me, and there are other people having their own side conversations and/or occasionally commenting 2-15 meters away, and other users even further away that I don't want to hear at all. 

In this situation I don't feel like I have a configuration option where: 

- the people within 2 meters of me are at full volume

- those 2-15 meters away are reduced so they don't heavily impact my ability to understand what's being said by the closest users but I can still make out what they're saying if I want to

- anybody further than 15 meters away is completely muted.


A wide transition range allows this behavior in a way that feels like for me it would be immersive and easy to tune on the fly.

I'm not suggesting changing the defaults, and I wouldn't mind if it's only possible with a manual value entry since it's not the "intended" behavior.

Also note that I'm suggesting 15 meters as a maximum settable value, not the default or suggested value.

**Comment by Ryn-Fox** [comment]
Also I know that there is already audio falloff in the game, but I've always found it too gradual and far carrying because I have very sensitive ears, audio processing issues, and executive dysfunction so separating conversations is incredibly difficult for me unless there's very little stimuli or a significant difference in volume.

**Comment by Frooxius** [comment]
Thank you for the additional context, that helps with this a fair bit.

The reason I ask is because anytime we get a request like that goes a bit against the design of something, I want to stop and take a step back to better understand the problem and see if there's a better solution.

I understand you're not asking for defaults to change. But part of this is to understand why we'd be making certain changes and see if there's a better way to approach the problem. In part, because I see some potential issues:

- If we increase the maximum, people might assume that setting the value high is expected since they'll see the large slider range, but that will make the earmuff less effective for them, without knowing why
- If we make it so you can only type it in, we're effectively fixing this mostly for you - this has poor discoverability for other users who might be facing same problem as you do
- It might not even solve the problem for you in the first place - have you tested this by forcing the value to be large? With this mechanism, people who are still quite close won't be attenuated as much

Based on your description what I think could work the best is adding an optional second "cascade" to the earmuff. That is you define initial zone and its volume, then you define another zone and its volume and then finally the "far" zone. Essentially it's just adding extra "layer" that you can control.

That way you could define first range to be 2 meters and it's volume to 100 % and the second range to be between 2-15 meters and its volume and then volume for everything further than that.

Would something like that work for you?

**Comment by Ryn-Fox** [comment]
I appreciate the detailed and carefully considered response. Those are a lot of good points. 

The proposed solution would meet my overall accessibility ask in a way that addresses discoverability and usage intent nicely, and is above and beyond my expectations!

I imagine each of those zones would also transition attenuation so volume doesn't abruptly change in a jarring manner?

**Comment by Frooxius** [comment]
Yes. There would be transition range for going from the first to second and then another for going from the second to "outside".

**Comment by Ryn-Fox** [comment]
Then I'm very much satisfied with the proposed solution and really excited to see it implemented.
